dotCMS/src/main/java/com/dotmarketing/beans/transform/TreeTransformer.java

```diff
@@ -9,33 +10,29 @@
 import com.google.common.collect.Lists;
 
 
-public class DBTreeTransformer{
+public class TreeTransformer implements DBTransformer<Tree> {
     
     private final List<Tree> trees;
     
-    public DBTreeTransformer (List<Map<String, Object>>list ){
+    public TreeTransformer(List<Map<String, Object>>list ){
         this.trees =  list
         .stream()
-        .map(row -> toTree(row))
+        .map(this::toTree)
         .collect(Collectors.toList());
-
     }
-    
-    public Tree tree() {
-        return trees.stream().findFirst().orElse(null);
-       
+
+    public TreeTransformer(Map<String, Object> map){
+        this.trees = Lists.newArrayList(toTree(map));
     }
-    public List<Tree> trees() {
+
+    @Override
+    public List<Tree> asList() {
         return trees;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Return a copy of "trees". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2384)